### DISCUSSION TITLE
Next.js API routes runs on a single thread

### DISCUSSION THREAD

**Discussion by Rahul-Vignesh** [vercel/next.js#20466] [Help]
When I put my application to production, all the API routes are handled by only one process out of the 7 processes (PM2 cluster mode). How can I balance the load?

**Comment by ctrlaltdylan** [comment]
Just bumping this thread because this is an important aspect to scaling the API portion of NextJS.

I haven't found any documentation or references to discussions about clustering API endpoints or implementing forky or throng to scale up a true full stack NextJS instance.

I was under the impression NextJS's API routing solution was production ready. I'm getting concerned that eventually this lack of clustering support will lead to a lot of over whelmed apps & maintainers.

**Reply by ctrlaltdylan** [reply]
Replying to myself because it sounds like you can get away with using pm2 with cluster mode to enable concurrency. I'm going to try and set this up to avoid doing it in app code.

**Reply by Rahul-Vignesh** [reply]
Actually I did try pm2 cluster mode on my production app, Something weird was happening where only one of processes were handling all the API routes.
